elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -1497,10 +1498,17 @@ protected void setValue(String fieldName, Object value) {
         Collection<Method> methods = dictionary.getTriggers(targetClass, annotationClass, fieldName);
         for (Method method : methods) {
             try {
-                method.invoke(obj);
+                method.invoke(obj, requestScope);
+            } catch (IllegalArgumentException exception) {
+                try {
+                    method.invoke(obj);
```

Comment:
Instead of guess and check by invocation, you can actually look at the signature of the method and see if it takes a RequestScope.